frontend/app/src/pages/home/join.tsx

```diff
@@ -1,11 +1,40 @@
+import { useState } from "react";
 import { Page } from "@/components";
 import { FindingMatchCard, SelectPreferencesCard } from "@/features/matching";
+import { io } from "socket.io-client";
+import { useNavigate } from "react-router-dom";
+import { Preferences } from "@/types/room";
 
 function JoinPage() {
+  const navigate = useNavigate();
+  const socket = io("http://localhost:8004");
+  const [waiting, setWaiting] = useState<boolean>(false);
+
+  const joinRoom = (preferences: Preferences) => {
+    setWaiting(true);
+    socket.emit(
+      "join",
+      preferences,
+      (response: { status: string; roomId: string }) => {
+        if (response.status === "joined") {
+          navigate(`room/${response.roomId}`);
```

Comment:
can add `room` to the list of route constants in `frontend/app/src/constants/routes.ts`